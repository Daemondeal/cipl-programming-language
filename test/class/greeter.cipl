class Greeter:
    init(greeting):
        this.greeting = greeting

    greet():
        println(this.greeting)

Greeter("Hello World").greet()
Greeter("Hi there").greet()