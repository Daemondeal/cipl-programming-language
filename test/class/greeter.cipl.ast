(class Greeter
    (proc init (params greeting) (body
        (expression (set greeting greeting this))
    ))
    (proc greet (body
        (expression (call println (get greeting this)))
    ))

)
(expression (call (get greet (call Greeter Hello World))))
(expression (call (get greet (call Greeter Hi there))))

