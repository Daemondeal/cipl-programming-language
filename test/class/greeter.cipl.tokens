Token          Lexeme         Line 
CLASS          class          1    
IDENTIFIER     Greeter        1    
COLON          :              1    
LINE_END       \n             2    
INDENT                        2    
IDENTIFIER     init           2    
LEFT_PAREN     (              2    
IDENTIFIER     greeting       2    
RIGHT_PAREN    )              2    
COLON          :              2    
LINE_END       \n             3    
INDENT                        3    
THIS           this           3    
DOT            .              3    
IDENTIFIER     greeting       3    
EQUAL          =              3    
IDENTIFIER     greeting       3    
LINE_END       \n             4    
DEDENT                        5    
IDENTIFIER     greet          5    
LEFT_PAREN     (              5    
RIGHT_PAREN    )              5    
COLON          :              5    
LINE_END       \n             6    
INDENT                        6    
IDENTIFIER     println        6    
LEFT_PAREN     (              6    
THIS           this           6    
DOT            .              6    
IDENTIFIER     greeting       6    
RIGHT_PAREN    )              6    
LINE_END       \n             7    
DEDENT                        8    
DEDENT                        8    
IDENTIFIER     Greeter        8    
LEFT_PAREN     (              8    
STRING         "Hello World"  8     Hello World
RIGHT_PAREN    )              8    
DOT            .              8    
IDENTIFIER     greet          8    
LEFT_PAREN     (              8    
RIGHT_PAREN    )              8    
LINE_END       \n             9    
IDENTIFIER     Greeter        9    
LEFT_PAREN     (              9    
STRING         "Hi there"     9     Hi there
RIGHT_PAREN    )              9    
DOT            .              9    
IDENTIFIER     greet          9    
LEFT_PAREN     (              9    
RIGHT_PAREN    )              9    
LINE_END       \n             9    
EOF                           9    
