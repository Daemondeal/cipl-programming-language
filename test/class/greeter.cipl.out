::stdout::
Hello World
Hi there

::stderr::
